test/flake.nix

```diff
@@ -0,0 +1,71 @@
+{
+  description = "";
+
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+  };
+
+  outputs = { self, nixpkgs, rust-overlay }:
+    let
+      supportedSystems = [ "x86_64-linux" ];
+      forAllSystems = nixpkgs.lib.genAttrs supportedSystems;
+      overlays = [
+        rust-overlay.overlays.default
+        self.overlay
+      ];
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system overlays; });
+
+    in {
+      overlay = final: prev: {
+      };
+
+      packages = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+          lib = pkgs.lib;
+        in rec {
+          inherit pkgs;
+
+          initrd = pkgs.makeInitrdNG {
+            name = "aster-initrd";
+            compressor = "gzip";
+
+            contents = let
+              initrdBinEnv = pkgs.buildEnv {
+                name = "initrd-bin-env";
+                paths = [
+                  pkgs.busybox
+                  pkgs.python3
+                ];
+                pathsToLink = ["/bin" "/sbin"];
```

Comment:
In Nix, all packages are stored in `/nix/store`.

Firstly, `pkgs.busybox` is a package, which is stored in `/nix/store/<hash>-busybox-1.36.1`, and its binaries, for example, `busybox` is placed on `/nix/store/<hash>-busybox-1.36.1/bin/busybox`.

Next, we created a environment with `pkgs.buildEnv`, then the environment will be stored in `/nix/store/<hash>-initrd-bin-env`. And it will symlink every `pathsToLink` of every `paths` to its path, that is,
```
/nix/store/<hash>-initrd-bin-env/bin/busybox  ->  /nix/store/<hash>-busybox-1.36.1/bin/busybox
/nix/store/<hash>-initrd-bin-env/bin/sh       ->  /nix/store/<hash>-busybox-1.36.1/bin/sh
...
```

Finally, we symlink initrd `/bin` to `${initrdBinEnv}/bin` with `"/bin".source = "${initrdBinEnv}/bin";`, that is,
```
/bin  ->  /nix/store/<hash>-initrd-bin-env/bin
```

So after that, executing `/bin/sh` will execute `/nix/store/<hash>-busybox-1.36.1/bin/sh` with symbolic link resolving.

> `pkgs.python3` is added here accidentally, and will be removed later.

Reply:
Thanks for your explanation!